### PR TITLE
Fix regression loss does not support empty tensor problem.

### PR DESCRIPTION
**Pull request by hhaAndroid**
## Motivation

Fix YOLOX l1loss assert error when the number of positive samples is 0.

resolve: https://github.com/open-mmlab/mmdetection/issues/5966

## Modification
When the positive sample is 0, the l1loss function returns 0.

### CODE REVIEW

**Comment by hhaAndroid** [comment]
Note: In fact, all L1Loss series have this problem, should it be processed directly in the loss function instead of in the YOLOX algorithm?

**Comment by ZwwWayne** [comment]
> Note: In fact, all L1Loss series have this problem, should it be processed directly in the loss function instead of in the YOLOX algorithm?

Such a fix might be better.

**Review by RangiLyu — APPROVED** [review]
LGTM

**Comment by ZwwWayne** [comment]
> > Note: In fact, all L1Loss series have this problem, should it be processed directly in the loss function instead of in the YOLOX algorithm?
> 
> Such a fix might be better.

Still suggest fixing it in L1Loss rather than in the head.

**Review by RangiLyu — APPROVED** [review]

File: mmdet/models/losses/iou_loss.py [hunk 1]
```diff
@@ -88,6 +91,7 @@ def bounded_iou_loss(pred, target, beta=0.2, eps=1e-3):
                             (target_w + eps))
     loss_dh = 1 - torch.min(target_h / (pred_h + eps), pred_h /
                             (target_h + eps))
+    # view(..., -1) does not work for empty tensor
     loss_comb = torch.stack([loss_dx, loss_dy, loss_dw, loss_dh],
                             dim=-1).view(loss_dx.size(0), -1)
```
Comment (ZwwWayne) [inline mmdet/models/losses/iou_loss.py:96]: We can use flatten rather than view to avoid that

**Review by ZwwWayne — APPROVED** [review]
